inc/classes/optimization/CSS/class-combine.php

```diff
@@ -30,89 +29,64 @@ class Combine extends Abstract_CSS_Optimization {
 	 * @since 3.1
 	 * @author Remy Perona
 	 *
-	 * @param HtmlPageCrawler $crawler  Crawler instance.
-	 * @param Options         $options  Options instance.
-	 * @param Minify\CSS      $minifier Minifier instance.
+	 * @param Options    $options  Options instance.
+	 * @param Minify\CSS $minifier Minifier instance.
 	 */
-	public function __construct( HtmlPageCrawler $crawler, Options $options, Minify\CSS $minifier ) {
-		parent::__construct( $crawler, $options );
+	public function __construct( Options $options, Minify\CSS $minifier ) {
+		parent::__construct( $options );
 
 		$this->minifier = $minifier;
 	}
 
 	/**
-	 * Combines multiple Google Fonts links into one
+	 * Minifies and combines all CSS files into one
 	 *
 	 * @since 3.1
 	 * @author Remy Perona
 	 *
+	 * @param string $html HTML content.
 	 * @return string
 	 */
-	public function optimize() {
-		$nodes = $this->find( 'link[href*=".css"]' );
+	public function optimize( $html ) {
+		$styles = $this->find( '<link\s+([^>]+[\s\'\"])?href\s*=\s*[\'\"]\s*?([^\'\"]+\.css(?:\?[^\'\"]*)?)\s*?[\'\"]([^>]+)?\/?>', $html );
```

Comment:
No need to escape double quotes: `[\s\'\"]`, `[\'\"]`, `[^\'\"]` => `[\s"\']`, `["\']`, `[^"\']`.